langchain/document_loaders/parsers/audio.py

```diff
@@ -0,0 +1,19 @@
+
+from typing import Iterator
+from langchain.document_loaders.base import BaseBlobParser
+from langchain.document_loaders.blob_loaders import Blob
+from langchain.schema import Document
+
+class OpenAIWhisperParser(BaseBlobParser):
```

Comment:
@eyurtsev may be helpful to discuss this briefly in case I misunderstood. I followed the logic of other parsers, but I also notice that this will reproduce much of the logic we already have in the document loader. Perhaps this is OK because the parser operates on a `blob` whereas the document loader is initialized w/ a `audio_file_path`. But, I'd like to be sure I understand the difference.

Reply:
The role of the parser is to make sure that we can implement parsing/transformation logic on raw bytes without caring about where the raw bytes came from (they can be in memory or on disk or on s3 or on a website).

```python
import openai
from langchain.document_loaders.generic import GenericLoader

class OpenAIWhisperParser(BaseBlobParser):
    """Transcribe and parse  audio files using audio-to-text transcription with OpenAI Whisper model."""

    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
        """Lazily parse the blob."""
        import openai
        
        with blob.as_bytes_io() as f:
            transcript = openai.Audio.transcribe('whisper-1', f)
        yield Document(page_content=transcript.text,metadata={"source": blob.source})

loader = GenericLoader.from_filesystem('directory', glob="*.mp3", parser=OpenAIWhisperParser())
docs = loader.load()
```

In principle, we don't even need to introduce a new loader since the GenericLoader can handle this.

One inconvenience is that at the moment it only supports loading patterns from the file system, so we can augment it to pick up a blob from a specific filepath 

`GenericLoader.from_path()` or something of this sort

Reply:
> One inconvenience is that at the moment it only supports loading patterns from the file system, so we can augment it to pick up a blob from a specific filepath

imo `glob` for `mp3 files` is OK since a specific file name can always be added to the regex.